### ISSUE TITLE
IconButton tooltip should appear on keyboard focus

### ISSUE THREAD

**Issue by marcianx** [flutter/flutter#128292]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

This is mostly focused on Flutter-web use-cases and thus references the ARIA spec.

1. Run the first `IconButton` example with a tooltip: https://api.flutter.dev/flutter/material/IconButton-class.html.
2. Hover the mouse. The tooltip appears.
3. Move the mouse away and tab-focus into the button. No tooltip appears.

### Expected results

While it's true that browser-native tooltips do not appear on button focus, [the ARIA spec for Web accessibility requires tooltips to appear on keyboard focus and hide when `Escape` is pressed](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role). This helps sighted users with physical impairments who cannot use the mouse be able to know what an icon button does.

### Actual results

The tooltip doesn't appear on `IconButton` focus.

### Code sample

 https://api.flutter.dev/flutter/material/IconButton-class.html

### Flutter Doctor output

N/A

**Comment by goderbauer** [comment]
> I'm assuming we'd want the same behavior on AOT desktop

This needs to be verified first whether that's expected/native behavior.

**Comment by chunhtai** [comment]
The tooltip is already sent to the engine https://api.flutter.dev/flutter/semantics/SemanticsProperties/tooltip.html

There may be work needed in the web engine if it doesn't wire it up

**Comment by yjbanov** [comment]
@chunhtai This issue is not with the semantics tree, but widget behavior. The request is that when the widget gains focus via tab traversal, it shows the tooltip. Currently, mouse hover is the only way to cause the tooltip to show up.

**Comment by chunhtai** [comment]
If we want to support this, the focusNode needs to be able to tell how the focus is received.

Hi @gspencergoog Do you know if this is possible? or is this something that you have came cross?

**Comment by gspencergoog** [comment]
No, the focus system doesn't know how focus was received: there's only `requestFocus`. Why does it need to know how the focus was received? Wouldn't the tooltip appear no matter how it received focus?

It's also not really clear how to define "traversal", since developers could wire lots of things up to change focus.
